firestore/query.go

```diff
@@ -304,6 +304,15 @@ func (q Query) Deserialize(bytes []byte) (Query, error) {
 	return q.fromProto(&runQueryRequest)
 }
 
+// NewAggregationQuery returns an AggregationQuery with this query as its
+// base query.
+func (q *Query) NewAggregationQuery() *AggregationQuery {
+	return &AggregationQuery{
+		query:            q,
+		aggregateQueries: make([]*pb.StructuredAggregationQuery_Aggregation, 0),
```

Comment:
Since you're not allocating a certain length, is there a need to make a slice at all? The first append below would do this for you.

Reply:
Done.